static/src/javascripts/projects/common/modules/commercial/build-page-targeting.js

```diff
@@ -299,13 +302,21 @@ const buildPageTargetting = (
     // This can be removed once we get sign-off from third parties who prefer to use appNexusPageTargeting.
     page.pageAdTargeting = pageTargeting;
 
-    return pageTargeting;
+    return Promise.resolve(pageTargeting);
 };
 
-const getPageTargeting = (): { [key: string]: mixed } => {
-    if (Object.keys(myPageTargetting).length !== 0) return myPageTargetting;
+const getPageTargeting = async (): Promise<PageTargettingLoose> => {
```

Comment:
I think this function can be simplified:

```js
const getPageTargeting = (): Promise<PageTargettingLoose> =>
    new Promise(resolve => {
        if (Object.keys(myPageTargetting).length !== 0)
            return resolve(myPageTargetting);

        onConsentChange(async state => {
            let canRun: boolean | null;
            if (state.ccpa) {
                // CCPA mode
                canRun = !state.ccpa.doNotSell;
            } else if (state.tcfv2) {
                // TCFv2 mode
                canRun = state.tcfv2.consents
                    ? Object.keys(state.tcfv2.consents).length > 0 &&
                      Object.values(state.tcfv2.consents).every(Boolean)
                    : false;
            } else if (state.aus) {
                // AUS mode
                canRun = state.aus.personalisedAdvertising;
            } else canRun = false;

            if (canRun !== latestConsentCanRun) {
                const ccpaState = state.ccpa ? state.ccpa.doNotSell : null;
                const eventStatus = state.tcfv2
                    ? state.tcfv2.eventStatus
                    : 'na';
                myPageTargetting = await buildPageTargetting(
                    canRun,
                    ccpaState,
                    eventStatus
                );
                latestConsentCanRun = canRun;
                resolve(myPageTargetting);
            }
        });
    });
```

Reply:
I’m thinking we could introduce `getConsentFor('framework', state)` or `getConsentBoolean(state)`. 

Looking at how other consumers of the CMP are using it, the granularity of TCFv2 is rarely leveraged.